src/snowflake/connector/query_context_cache.py

```diff
@@ -180,91 +169,119 @@ def _get_elements(self) -> set[QueryContextElement]:
 
     def _last(self) -> QueryContextElement:
         return self._tree_set[-1]
-
-    def deserialize_from_arrow_base64(self, data: str) -> None:
-        if not installed_pandas:
-            self._data = data
-            return
-
+    
+    def serialize_to_json(self) -> str:
         with self._lock:
-            logger.debug(
-                f"deserialize_from_arrow_base64() called: data from server: {data}"
-            )
-            self.log_cache_entries()
-
-            if data is None or len(data) == 0:
-                self.clear_cache()
-                logger.debug("deserialize_from_arrow_base64() returns")
-                self.log_cache_entries()
-                return
-
-            decoded_data = b64decode(data)
-            input = BytesIO(decoded_data)
-            try:
-                with pa.ipc.open_stream(input) as reader:
-                    for record_batch in reader:
-                        record_dict = record_batch.to_pydict()
-                        for i in range(len(record_batch)):
-                            self.merge(
-                                record_dict["id"][i],
-                                record_dict["timestamp"][i],
-                                record_dict["priority"][i],
-                                record_dict["context"][i],
-                            )
-            except Exception as e:
-                logger.debug(f"deserialize_from_arrow_base64: Exception = {e}")
-                # clear cache due to incomplete merge
-                self.clear_cache()
-
-            self.check_cache_capacity()
-            logger.debug("deserialize_from_arrow_base64() returns")
-            self.log_cache_entries()
-
-    def serialize_to_arrow_base64(self) -> str:
-        if not installed_pandas:
-            return self._data
-
-        with self._lock:
-            logger.debug("serialize_to_arrow_base64() called")
+            logger.debug("serialize_to_json() called")
             self.log_cache_entries()
 
             if len(self._tree_set) == 0:
                 return None
 
             try:
-                stream = BytesIO()
                 size = self.get_size()
                 id_vals = [None] * size
                 timestamp_vals = [None] * size
                 priority_vals = [None] * size
                 context_vals = [None] * size
                 self.get_elements(id_vals, timestamp_vals, priority_vals, context_vals)
```

Comment:
Do we still need to use 4 elements arrays? I think We probably can just use the `qce` object itself.

Reply:
I think it is OK to stay with the original implementation here.

Reply:
I think this introduced unnecessary complications to the code so I'd like to see it optimized. Or at the very least we should create a JIRA and fix it in a follow-up PR.

Reply:
I see. Let me try to optimize it in this PR then.